src/utils.ts

```diff
@@ -1,12 +1,15 @@
-import { assert, pattern, string } from 'superstruct';
+import { assert, define } from 'superstruct';
 import type { Struct } from 'superstruct';
 
+const UUID_V4_REGEX =
+  /^[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/iu;
+
```

Comment:
is the `u` required? UUIDs only contain ASCII characters, so I don't think the unicode flag is required but it also does not harm to have it

(Just a comment to get your opinion, not asking for you to change it)